docs/releasing/publishing-a-pre-release.md

```diff
@@ -0,0 +1,135 @@
+
+# Publish a new version of GOV.UK Frontend
+
+Developers should pair on pre-releases. When remote working, it can be useful to be on a call together.
+
+1. Check out the **main** branch and pull the latest changes.
+
+2. Ensure you're running the version of NodeJS matching [`.nvmrc`](/.nvmrc).
+
+   - If you use NVM, run `nvm use` to set up the right version
+   - If you use another management system (like [`asdf`](https://asdf-vm.com/guide/getting-started.html)), compare the output of `node --version` and install the right one if necessary
+
+3. Run `npm ci` to make sure you have the exact dependencies installed.
+
+4. Determine the pre-release identifier
+
+   - Use `internal` for internal pre-releases
+   - Use `beta` for beta pre-releases
+
+5. Determine the pre-release version type
+
+   As examples with a `beta` pre-release identifier:
+
+   - Use `premajor` to bump from `v4.7.0` to `v5.0.0-beta.0`
+   - Use `preminor` to bump from `v4.7.0` to `v4.8.0-beta.0`
+   - Use `prepatch` to bump from `v4.7.0` to `v4.7.1-beta.0`
+
+   See the [versioning documentation](/docs/contributing/versioning.md) for more information.
+
+   Alternatively, when publishing an update to an existing pre-release:
+
+   - Use `prerelease` to bump from `v5.0.0-beta.0` to `v5.0.0-beta.1`
+   - Use `prerelease` to move from `v5.0.0-internal.X` to `v5.0.0-beta.0`.
+     Note that this resets the final number to `.0`, so prefer setting the version manually if a pre-release of the kind you're running has already gone through.
+
+6. Apply the new pre-release version number by running:
+
+   ```shell
+   npm version <PRE-RELEASE TYPE> --preid <PRE-RELEASE IDENTIFIER> --no-git-tag-version --workspace govuk-frontend
+   ```
+
+   This step will update [`govuk-frontend`'s `package.json`](/packages/govuk-frontend/package.json) and project [`package-lock.json`](/package-lock.json) files.
+
+   Do not commit the changes.
+
+7. Create and check out a new branch (`release-[version]`)
+
+   ```shell
+   git switch -c "release-$(npm run version --silent --workspace govuk-frontend)"
+   ```
+
+8. If you're publishing a beta pre-release, update the [`CHANGELOG.md`](/CHANGELOG.md) by:
+
+   - changing the 'Unreleased' heading to the new version number and release type. For example, '5.0.0-beta.0 (Pre-release)'
+   - adding a new 'Unreleased' heading above the new version number and release type, so users will know where to add PRs to the changelog
+   - saving your changes
+
+9. Run `npm run build-release` to:
+
+   - build GOV.UK Frontend into [the package's `/dist`](/packages/govuk-frontend/dist) and [root `/dist`](/dist) directories
+   - commit the changes
+   - push a branch to GitHub
+
+   You will now be prompted to continue or cancel.
```

Comment:
It might seem pedantic, but should we give the user an action here? Similar to step 5 of "Publish a release to npm"

Reply:
I'm not sure we can here. In the npm part, step 4, we can guide the users towards pressing `N` as we know they'll have to change from 'latest' to 'next' or 'internal'. This is more like the step 5 of npm where we need the person driving to have a pause and consider whether to continue, rather than automatically go "'y', let's go!"

Reply:
Oh, I see, yeah, I can add the same clarification here, good shout 😊